dotnet/csharp.bzl

```diff
@@ -48,50 +48,55 @@ def _get_libdirs(dlls, libdirs=[]):
   return [dep.dirname for dep in dlls] + libdirs
 
 def _make_csc_arglist(ctx, output, depinfo, extra_refs=[]):
+  args = ctx.actions.args()
+
   flag_start = ctx.attr._flag_start
-  args = [
-       # /out:<file>
-      _make_csc_flag(flag_start, "out", output.path),
-       # /target (exe for binary, library for lib, module for module)
-      _make_csc_flag(flag_start, "target", ctx.attr._target_type),
-      # /fullpaths
-      _make_csc_flag(flag_start, "fullpaths"),
-      # /warn
-      _make_csc_flag(flag_start, "warn", str(ctx.attr.warn)),
-      # /nologo
-      _make_csc_flag(flag_start, "nologo"),
-  ]
+
+  # /out:<file>
+  args.add(_make_csc_flag(flag_start, "out", output.path))
+
+  # /target (exe for binary, library for lib, module for module)
+  args.add(_make_csc_flag(flag_start, "target", ctx.attr._target_type))
+
+  # /fullpaths
+  args.add(_make_csc_flag(flag_start, "fullpaths"))
+
+  # /warn
+  args.add(_make_csc_flag(flag_start, "warn", str(ctx.attr.warn)))
+
+  # /nologo
+  args.add(_make_csc_flag(flag_start, "nologo"))
 
   # /modulename:<string> only used for modules
   libdirs = _get_libdirs(depinfo.dlls)
   libdirs = _get_libdirs(depinfo.transitive_dlls, libdirs)
 
   # /lib:dir1,[dir1]
   if libdirs:
-    args += [_make_csc_flag(flag_start, "lib", ",".join(list(libdirs)))]
+    args.add(libdirs, format=flag_start + "lib:%s")
```

Comment:
I kept `flag_start` for consistency but didn't use `_make_csc_flag` (which otherwise deals with `:`) in the next PR I'll switch to `/` across the board and drop `_make_csc_flag` because it's not pulling it's weight.

Reply:
Thanks! Did you try if it works?
For what I can tell, this looks good.

Reply:
Yep, sorry forgot to mention: tried it on Mac and Linux with Mono.

On that note: what would be the process for setting up CI for this repo? What is the preferred service for that (Travis? A Jenkins service you run? etc.) and would it work if I PR'd the stuff assuming a maintainer will click the buttons I can't to turn stuff on?

I can look around at other rules repos to get a sense of how to organize the tests but if this is something that makes sense for contribution and you have pointers that's cool too.

Reply:
Oh neat, I didn't notice. Is it set up to check PRs? I don't see any checks in this PR but maybe it only posts on fail?

Reply:
Judging by https://ci.bazel.io/job/rules_dotnet/GitHubPollLog/ it's set to trigger on master chages only. I'd be cool to have it in PRs so these changes we're safer to review.

Reply:
Honestly, I don't know. If I understand the Jenkins UI correctly, there's a [directory for PR jobs](https://ci.bazel.io/view/Bazel%20bootstrap%20and%20maintenance/job/PR/), which I suppose are the tests that Jenkins runs on a PR or Gerrit review. But all runs are weeks or months old, so this might be turned of, or not running as presubmit, or both.

Let me pull in @philwo who started working on CI today and might know more.

Reply:
Actually, maybe I'm wrong and this CI job is supposed to test Bazel changes with the `rules_dotnet` depot, not the changes in the `rules_dotnet` depot itself.

Reply:
I guess this was supposed to work, but maybe it doesn't? It surely would be very useful. I wonder if we can just trigger it using the magic words? Let me try this.

I'm working on improving our CI, but it will be a few weeks until we have something to show, as the only person who built and maintained the current system left the team recently and we have to figure out how this all works first.